bodo/pandas/plan.py

```diff
@@ -43,6 +43,11 @@ def __init__(self, plan_class, empty_data, *args):
 
         self.pa_schema = pa.Schema.from_pandas(self.empty_data)
 
+    def _update_column_names(self, new_cols):
+        """Update the column names by modifying empty_data and pa_schema"""
+        self.empty_data.columns = new_cols
```

Comment:
Is there a presumption here that len(new_cols) matches the current number of columns in the plan?  Error check this?